homeassistant/helpers/entity_registry.py

```diff
@@ -1313,6 +1335,18 @@ def async_entries_for_label(
     return [entry for entry in registry.entities.values() if label_id in entry.labels]
 
 
+@callback
+def async_entries_for_category(
+    registry: EntityRegistry, scope: str, category_id: str
+) -> list[RegistryEntry]:
+    """Return entries that match a category in a scope."""
+    return [
+        entry
+        for entry in registry.entities.values()
+        if scope in entry.categories and entry.categories[scope] == category_id
```

Comment:
Same

Reply:
Addressed in [deb45eb](https://github.com/home-assistant/core/pull/110897/commits/deb45eb3cbb1137b50322a63073bcbb6eb20e44a)